.github/workflows/platform.dependencies.yml

```diff
@@ -286,6 +291,146 @@ jobs:
           managementGroupId: '${{ secrets.ARM_MGMTGROUP_ID }}'
           removeDeployment: '${{ env.removeDeployment }}'
 
+  job_deploy_imgt:
+    runs-on: ubuntu-20.04
+    name: 'Deploy image template'
+    if: github.event.inputs.deployVhdDependencies == 'true'
+    env:
+      namespace: 'Microsoft.VirtualMachineImages\imageTemplates'
+    needs:
+      - job_deploy_rolea
+      - job_deploy_sig
+    outputs:
+      imageTemplateName: ${{ steps.print_imgt_output.outputs.imageTemplateName }}
+      imageTemplateResourceGroup: ${{ steps.print_imgt_output.outputs.imageTemplateResourceGroup }}
+    strategy:
+      fail-fast: false
+      matrix:
+        parameterFilePaths: ['parameters.json']
+    steps:
+      - name: 'Checkout'
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: 'Deploy module'
+        id: deploy_imgt
+        uses: ./.github/actions/templates/validateModuleDeployment
+        with:
+          templateFilePath: 'arm/${{ env.namespace }}/deploy.bicep'
+          parameterFilePath: '${{ env.dependencyPath }}/${{ env.namespace }}/parameters/${{ matrix.parameterFilePaths }}'
+          location: '${{ env.defaultLocation }}'
+          resourceGroupName: '${{ env.defaultResourceGroupName }}'
+          subscriptionId: '${{ secrets.ARM_SUBSCRIPTION_ID }}'
+          managementGroupId: '${{ secrets.ARM_MGMTGROUP_ID }}'
+          removeDeployment: '${{ env.removeDeployment }}'
+      - name: 'Set image template output'
+        id: print_imgt_output
+        uses: azure/powershell@v1
+        with:
+          inlineScript: |
+            $deploymentOutput = '${{ steps.deploy_imgt.outputs.deploymentOutput }}'
+            $imageTemplateName = (ConvertFrom-Json $deploymentOutput).name
+            $imageTemplateResourceGroup = (ConvertFrom-Json $deploymentOutput).resourceGroupName
+            Write-Verbose "imageTemplateName: $imageTemplateName" -Verbose
+            Write-Verbose "imageTemplateResourceGroup: $imageTemplateResourceGroup" -Verbose
+            Write-Output ('::set-output name={0}::{1}' -f 'imageTemplateName', $imageTemplateName)
+            Write-Output ('::set-output name={0}::{1}' -f 'imageTemplateResourceGroup', $imageTemplateResourceGroup)
+          azPSVersion: 'latest'
+
+  job_deploy_vhd_to_sa:
```

Comment:
Would it make sense to merge these steps to the image template deployment? Would avoid the need to pass outputs around

Reply:
Outputs would need to be passed even between steps. The main reasons why I've put it in a separate job are: 

1. It is consistent with KV and storage post deployment which are also in a separate job
2. Having them in the same job would mean running a double step for the agent setup. One as a step in the job and one as a step in the deployment action and I found that a bit messy to follow. Anyway we could discuss about it.